File: Data_Structures/JAVA/src/00Array/Array1.java
import java.util.Set;
import java.util.HashSet;
public class Array1 {

//time and space complexity = 0(n)TS
     public static int[] twoNumberSum(int[] array,int target){

       Set<Integer> nums = new HashSet<>(); 
         for(int i=0; i<array.length; i++){
            int num = array[i];
            int match = target-num;
            if(nums.contains(match)){
                return new int[]{num, match};
            }else{
              nums.add(num);
            }
         }
        return new int[0];
    }


    public static void main(String[] args) throws Exception {

         int[] array = {6,5,7,9,4,0,2};
         int[] result = twoNumberSum(array, 10);
         System.out.println("first:"+result[0]+"second:"+result[1]);
    }
    
}

File: Data_Structures/JavaScript/00Arrays/array1.js
// here is the same question we have done in the previous
//sumOfTwoNumbers with less complexiety
//In JavaScript, we use the Set data structure to store unique values,
// similar to the HashSet in Java.

const sumOfTwoNumbers = (array, target)=>{
      let set = new Set;
    for(let i = 0; i<array.length; i++){
        num = array[i];
        let match = target - array[i];
        if(set.has(match)){
            return [num,match];
        }
        set.add(num);
    }
    return [];
     
}


let array = [6, 5, 7, 8, 4, 0];
let result = sumOfTwoNumbers(array,10);
console.log(result);


//In JavaScript, the Set data structure is typically implemented as a 
//hash table. Checking if an element exists in a hash table 
//has an average time complexity of O(1), 
//meaning it takes constant time regardless of the size of the set.
